src/System.IO/src/System/IO/BufferedStream.cs

```diff
@@ -234,7 +236,11 @@ protected override void Dispose(bool disposing)
             finally
             {
                 _stream = null;
-                _buffer = null;
+                if (_buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
I think I understand what you're saying, but won't people need to expect to get already-used buffers when using the .Shared pool? It seems like it's part of the contract that rented buffers might not be cleared.

What unexpected consequences are you referring to?